### PR TITLE
feat(HACBS-2011): task push-snapshot can push a tag with git sha

### PR DESCRIPTION
**Pull request by johnbieren**
This PR adds the `addGitShaTag` parameter to the `push-snapshot` task. When true, a sha will be looked for in the snapshot for each component and a tag will be pushed with said sha when pushing to the external registry

The work to inject the git sha in the snapshot is not yet complete, so marking this as do not merge for now

### CODE REVIEW

**Comment by johnbieren** [comment]
This PR also reworks the push-snapshot task to use jq to extract JSON values instead of creating a bash array named `data`

File: catalog/task/push-snapshot/0.6/push-snapshot.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        }
+
+        application=$(jq -r '.application' <<<'$(params.mappedSnapshot)')
+        printf 'Beginning "%s" for "%s"\n\n' "$(context.task.name)" "$application"
+        for component in $(jq -rc '.components[]' <<< '$(params.mappedSnapshot)')
+        do
+          containerImage=$(jq -r '.containerImage' <<< $component)
+          repository=$(jq -r '.repository' <<< $component)
+          name=$(jq -r '.name' <<< $component)
+          git_sha=$(jq -r '.source.git.commit' <<< $component) # this sets the value to "null" if it doesn't exist
```
[the remark this answers is not in the document]
Comment (johnbieren) [reply catalog/task/push-snapshot/0.6/push-snapshot.yaml:64]: @scoheb fyi. switching back to revision

**Comment by johnbieren** [comment]
@theflockers @davidmogar @scoheb @happybhati any feedback?

**Comment by johnbieren** [comment]
https://github.com/redhat-appstudio/integration-service/pull/135 adds the field and is in review. Once it merges, do we want to turn the parameter to add the git sha tag to default to true @scoheb @davidmogar ?

**Comment by scoheb** [comment]
@johnbieren yes...default should be true.

**Review by scoheb — APPROVED** [review]

**Review by happybhati — APPROVED** [review]
LGTM, only the PR description needs to be updated before the merge.

> "The work to inject the git sha in the snapshot is not yet complete, so marking this as do not merge for now"

**Comment by johnbieren** [comment]
> @johnbieren yes...default should be true.

I am going to merge as is (with default false) and will open a new PR to the same version of the pipeline to change the default to true once integration service has their stuff merged

**Pull request merged**